nucypher/cli/characters/alice.py

```diff
@@ -117,10 +119,22 @@ def alice(click_config,
         if dev:
             raise click.BadArgumentUsage("Cannot create a persistent development character")
 
+        if not provider_uri and not federated_only:
+            raise click.BadOptionUsage(option_name='--provider',
+                                       message="--provider is required to create a new decentralized alice.")
+
         if not config_root:                         # Flag
             config_root = click_config.config_file  # Envvar
 
-        new_alice_config = AliceConfiguration.generate(password=get_password(confirm=True),
+        if not pay_with and not federated_only:
```

Comment:
Sorry - I dont mean to ignore any comments, just a bit of alert fatigue.  Backstory here:  originally the flag was `--checksum-address` for all characters.  This started to feel a little meaningless in regards to alice, who only uses the blockchain to pay for policies.  Bob CLI was developed last, and the CLI flag from alice was copied and pasted into bob, leaving `--pay-with` as a refactoring artifact which I initially accepted.  Bob's only reason to have a ethereum wallet, is to challenge incorrect re-encryptions.  Is it appropriate to call this `--pay-with`?

Reply:
No worries - I definitely didn't take it personally and figured you just missed it 😄 

It does seem like `pay-with` is a bit more descriptive than `checksum-address`. Perhaps `wallet-address` or something of the like could be good...?

Reply:
"pay with" was easy to name since it only does just that.  The other CLI modules have a more broad use of this flag across different actions.  However, I am certainly interested in an alternate name for `--checksum-address`.  At the moment this name reflects that of a character (`Character.checksum_address`), and may actually be a "federated checksum address"

Reply:
👍